### DISCUSSION TITLE
[ideas] Clarifications about the `iteration` argument

### DISCUSSION THREAD

**Discussion by multimeric** [ropensci/targets#1280] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I just hit an issue very similar to Elio's discussion here: https://github.com/ropensci/targets/discussions/1216. The current wording for iteration in the book seems to be:

> The iteration argument of `tar_target()` determines how to split non-dynamic targets and how to aggregate dynamic ones. 

I guess in Elio's example `paths_list` is the non-dynamic target that's being split, and hence its own `iteration` argument affects how it is split. 

Some of my questions about this argument that I'd love to have answered are:

* Does `iteration` affect the splitting/aggregation of the current target, or incoming targets? In Elio's example it seems to affect the current target. Is that always the case?
* Does `iteration` only come into effect at the boundary between a dynamic and static target? What about if one dynamic branched target is the input for another dynamic branched target? 

Once this is clarified, I'm happy to submit a PR to improve the docs here.

**Comment by Aariq** [comment]
I was also confused initially about which targets `iteration` affects, but I think I maybe get it now, and my mental model of what the argument does has changed a bit.  I now think of it as basically "how should we treat _this_ target—as a list or as a vector (or a grouped dataframe) when it needs to get split or its pieces need to be recombined?"

**Reply by multimeric** [reply]
Thanks, that certainly helps with my first question!

**Comment by wlandau** [comment, marked as answer]
> Does iteration only come into effect at the boundary between a dynamic and static target? What about if one dynamic branched target is the input for another dynamic branched target?

In that case, the downstream dynamic target always branches over the existing branch targets of the upstream target. Since the upstream branch targets already exist and do not need to be combined, the `iteration` argument of the upstream target is irrelevant for the downstream one.
